cupy/_core/internal.pyx

```diff
@@ -130,14 +130,28 @@ cdef void get_reduced_dims(
 @cython.profile(False)
 cdef inline Py_ssize_t get_contiguous_strides_inplace(
         const shape_t& shape, strides_t& strides,
-        Py_ssize_t itemsize, bint is_c_contiguous):
+        Py_ssize_t itemsize, bint is_c_contiguous, bint zeros_for_zerosize):
     cdef Py_ssize_t st, sh
     cdef Py_ssize_t is_nonzero_size = 1
     cdef int i, ndim = shape.size()
     cdef Py_ssize_t idx
     strides.resize(ndim, 0)
     st = 1
 
+    # For some cases, the strides are all set to zero if the shape is
+    # zero-sized since NumPy 1.23.
```

Comment:
Do you know whether NumPy intentionally kept the behavior for no copy reshape?  Maybe it's good to report to NumPy.

Reply:
No, I couldn't search the source enough last night. As you say, it's a kind of weird behavior. I'm going to see it again.

Reply:
It seems the change in creation (zero cleared strides) was introduced in https://github.com/numpy/numpy/pull/21477/.